daemon/containerd/image_load_test.go

```diff
@@ -0,0 +1,111 @@
+package containerd
+
+import (
+	"bytes"
+	"context"
+	"math/rand"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/containerd/containerd/content"
+	"github.com/containerd/containerd/content/local"
+	"github.com/containerd/containerd/namespaces"
+	"github.com/containerd/platforms"
+	"github.com/docker/docker/errdefs"
+	"github.com/docker/docker/internal/testutils/specialimage"
+	"github.com/docker/docker/pkg/archive"
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+	"gotest.tools/v3/assert"
+	is "gotest.tools/v3/assert/cmp"
+)
+
+func TestImageLoadMissing(t *testing.T) {
+	linuxAmd64 := ocispec.Platform{OS: "linux", Architecture: "amd64"}
+	linuxArm64 := ocispec.Platform{OS: "linux", Architecture: "arm64"}
+	linuxArmv5 := ocispec.Platform{OS: "linux", Architecture: "arm", Variant: "v5"}
+
+	ctx := namespaces.WithNamespace(context.TODO(), "testing-"+t.Name())
+
+	store, err := local.NewLabeledStore(t.TempDir(), &memoryLabelStore{})
+	assert.NilError(t, err)
+
+	imgSvc := fakeImageService(t, ctx, store)
+	// Mock the daemon platform.
+	imgSvc.defaultPlatformOverride = platforms.Only(linuxAmd64)
+
+	tryLoad := func(ctx context.Context, t *testing.T, dir string, platform ocispec.Platform) error {
+		tarRc, err := archive.Tar(dir, archive.Uncompressed)
+		assert.NilError(t, err)
+		defer tarRc.Close()
+
+		buf := bytes.Buffer{}
+
+		defer func() {
+			t.Log(buf.String())
+		}()
+
+		return imgSvc.LoadImage(ctx, tarRc, &platform, &buf, true)
+	}
+
+	clearStore := func(ctx context.Context, t *testing.T) {
+		assert.NilError(t, store.Walk(ctx, func(info content.Info) error {
+			return store.Delete(ctx, info.Digest)
+		}), "failed to delete all content")
+	}
+
+	t.Run("empty index", func(t *testing.T) {
+		imgDataDir := t.TempDir()
+		_, err := specialimage.EmptyIndex(imgDataDir)
+		assert.NilError(t, err)
+
+		err = tryLoad(ctx, t, imgDataDir, linuxAmd64)
+		assert.Check(t, is.Error(err, "image emptyindex:latest was loaded, but doesn't provide the requested platform (linux/amd64)"))
+		assert.Check(t, errdefs.IsNotFound(err))
+	})
+	clearStore(ctx, t)
+
+	t.Run("single platform", func(t *testing.T) {
+		imgDataDir := t.TempDir()
+		r := rand.NewSource(0x9127371238)
+		_, err := specialimage.RandomSinglePlatform(imgDataDir, linuxAmd64, r)
+		assert.NilError(t, err)
+
+		err = tryLoad(ctx, t, imgDataDir, linuxArm64)
+		assert.Check(t, is.ErrorContains(err, "doesn't provide the requested platform (linux/arm64)"))
+		assert.Check(t, errdefs.IsNotFound(err))
```

Comment:
Oh! Same comment here as on the other PR; https://github.com/moby/moby/pull/48722#discussion_r1810554194

> Perhaps use `is.ErrorType` - It's currently marked "deprecated", but will be un-deprecated in the next release; [gotestyourself/gotest.tools#286](https://github.com/gotestyourself/gotest.tools/pull/286)
> 
> ```go
> assert.Check(t, is.ErrorType(err, errdefs.IsNotFound))
> ```
> 
> Alternattively, add a custom comment to show the actual error-type and message;
> 
> ```go
> assert.Check(t, errdefs.IsNotFound(err), "expected  a ErrNotFound, got %[1]v (%[1]T)", err)
> ```

Reply:
Ah, good to heart that it will be undeprecated! ❤️

Reply:
Yeah, I understand the motivation (i.e., trying to keep `gotest.tools` as minimal as possible; don't duplicate assertions where there's no need), but the `is.ErrorType` was really much more convenient for our cases 😂